libs/render_delegate/utils.cpp

```diff
@@ -595,6 +595,33 @@ void HdArnoldSetRadiusFromPrimvar(AtNode* node, const SdfPath& id, HdSceneDelega
     AiNodeSetArray(node, str::radius, arr);
 }
 
+void HdArnoldSetNormalsFromPrimvar(AtNode* node, const SdfPath& id, const TfToken& primvarName, const AtString& arnoldAttr, HdSceneDelegate* sceneDelegate)
+{
+    HdArnoldSampledPrimvarType sample;
+    sceneDelegate->SamplePrimvar(id, primvarName, &sample);
+    HdArnoldSampledType<VtArray<GfVec3f>> xf;
+    HdArnoldUnboxSample(sample, xf);
+    if (xf.count == 0) {
+        return;
+    }
+
+    int timeIndex = 0;
+    for (size_t i = 0; i < xf.times.size(); ++i) {
+        if (xf.times[i] >= 0) {
+            timeIndex = i;
+            break;
+        }
+    }
+    const auto& v0 = xf.values[timeIndex];
+    AtArray* arr = AiArrayAllocate(v0.size(), xf.times.size(), AI_TYPE_VECTOR);
+    AtVector* out = static_cast<AtVector*>(AiArrayMap(arr));
```

Comment:
I think here we normally have to resample the values such that they are uniformly distributed in time before passing them to arnold. Unless it's done before which I didn't check.

Reply:
there is an example here https://github.com/Autodesk/arnold-usd/blob/fa3ca6194bfde47856ffe69a8b2af3decda052b8/libs/render_delegate/utils.cpp#L527

Reply:
Ported the code to the new function.